### PR TITLE
Environment variable improvements

### PR DESCRIPTION
**Pull request by nrwahl2**
Cleanup as part of T637

### CODE REVIEW

File: lib/common/logging.c [hunk 1]
```diff
@@ -798,6 +796,7 @@ set_identity(const char *entity, int argc, char *const *argv)
 
     CRM_ASSERT(crm_system_name != NULL);
 
+    // @COMPAT Drop at 3.0.0. Added via f3c4f9e without explanation. Unused.
     setenv("PCMK_service", crm_system_name, 1);
```
Comment (kgaillot) [inline lib/common/logging.c]: See fence-agents/lib/fencing.py.py

File: daemons/pacemakerd/pacemakerd.c [hunk 2]
```diff
@@ -425,10 +425,7 @@ main(int argc, char **argv)
 #endif
 
     if (pcmk__locate_sbd() > 0) {
-        setenv("PCMK_watchdog", "true", 1);
         running_with_sbd = TRUE;
-    } else {
-        setenv("PCMK_watchdog", "false", 1);
```
Comment (kgaillot) [inline daemons/pacemakerd/pacemakerd.c:430]: Interestingly, sbd/src/sbd-pacemaker.c also sets it, but sbd doesn't use it either

File: daemons/execd/pacemaker-execd.c [hunk 3]
```diff
@@ -514,7 +514,7 @@ main(int argc, char **argv, char **envp)
 
 #ifdef PCMK__COMPILE_REMOTE
     if (options.port != NULL) {
-        setenv("PCMK_remote_port", options.port, 1);
+        pcmk__set_env_option(PCMK__ENV_REMOTE_PORT, options.port);
```
Comment (kgaillot) [inline daemons/execd/pacemaker-execd.c:517]: I'd rather not introduce an `HA_` variant of environment variables that don't already have them. We can add an argument to `pcmk__set_env_option()` for that.

File: lib/common/watchdog.c [hunk 4]
```diff
@@ -89,13 +89,18 @@ panic_local(void)
 
     /* We're either pacemakerd, or a pacemaker daemon running as root */
 
-    if (pcmk__str_eq("crash", getenv("PCMK_panic_action"), pcmk__str_casei)) {
+    if (pcmk__str_eq("crash", pcmk__env_option(PCMK__ENV_PANIC_ACTION),
+                     pcmk__str_casei)) {
```
Comment (kgaillot) [inline lib/common/watchdog.c:93]: We should grab this in a variable

File: lib/common/remote.c [hunk 5]
```diff
@@ -186,8 +187,10 @@ get_bound_dh_bits(unsigned int dh_bits)
     int dh_min_bits;
     int dh_max_bits;
 
-    pcmk__scan_min_int(getenv("PCMK_dh_min_bits"), &dh_min_bits, 0);
-    pcmk__scan_min_int(getenv("PCMK_dh_max_bits"), &dh_max_bits, 0);
+    pcmk__scan_min_int(pcmk__env_option(PCMK__ENV_DH_MIN_BITS), &dh_min_bits,
+                       0);
+    pcmk__scan_min_int(pcmk__env_option(PCMK__ENV_DH_MIN_BITS), &dh_max_bits,
```
Comment (kgaillot) [inline lib/common/remote.c:192]: MAX

**Review by kgaillot — COMMENTED** [review]
nice

File: lib/common/logging.c (hunk 1 above)
Comment (nrwahl2) [reply lib/common/logging.c]: wow

wish we had a compendium of ClusterLabs users of obscure Pacemaker features

File: daemons/pacemakerd/pacemakerd.c (hunk 2 above)
Comment (nrwahl2) [reply daemons/pacemakerd/pacemakerd.c:430]: It's left over from an older implementation (see f557164 when it was dropped). Pacemaker didn't drop it when the implementation changed, so I'm not surprised that sbd didn't either :)

File: lib/common/watchdog.c (hunk 4 above)
Comment (nrwahl2) [reply lib/common/watchdog.c:93]: Good call

File: lib/common/remote.c (hunk 5 above)
Comment (nrwahl2) [reply lib/common/remote.c:192]: Whoops, thanks

File: daemons/execd/pacemaker-execd.c (hunk 3 above)
Comment (nrwahl2) [reply daemons/execd/pacemaker-execd.c:517]: Good point. I'll add a commit for this. My current plan is to ignore the new argument when we're *un*setting an option. The reason is that `pcmk__get_env_option()` checks for both, so it's better to make sure we fully clear it.

That's debatable though.

**Comment by nrwahl2** [comment]
Updated to address review

**Comment by nrwahl2** [comment]
One more time... Updated copyright in `daemons/execd/remoted_pidone.c` and used `PCMK__ENV_NODE_START_STATE` in one more place (new commit).

File: daemons/execd/pacemaker-execd.c (hunk 3 above)
Comment (kgaillot) [reply daemons/execd/pacemaker-execd.c]: Makes sense

File: daemons/pacemakerd/pacemakerd.c [hunk 6]
```diff
@@ -92,7 +92,7 @@ pid_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **er
 static gboolean
 standby_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **err) {
     options.standby = TRUE;
-    pcmk__set_env_option("node_start_state", "standby");
+    pcmk__set_env_option("node_start_state", "standby", true);
```
Comment (kgaillot) [inline daemons/pacemakerd/pacemakerd.c]: We can use false for this one, it was only documented as PCMK when it was added

**Review by kgaillot — COMMENTED** [review]
looks good

**Review by kgaillot — APPROVED** [review]

**Pull request merged**